php/libraries/Project.class.inc

```diff
@@ -0,0 +1,254 @@
+     */
+    private function __construct()
+    {
+    }
+
+    /**
+     * This returns a single Project object for the given
+     * project name, and will only ever instantiate one object in
+     * the event that it's called multiple times.
+     *
+     * @param string $projectName The project
+     *
+     * @note This should be used by the NDB_Factory and not be called
+     * directly
+     *
+     * @return Project the Project object
+     * @throws LorisException
+     */
+    static function &singleton($projectName)
+    {
+            if (!isset(self::$instances[$projectName])) {
+            $factory     = \NDB_Factory::singleton();
+            $projectData = $factory->database()->pSelectRow(
+                "
+                SELECT
+                  projectId,
+                  name,
+                  recruitmentTarget
+                FROM
+                  Project
+                WHERE name = :v_project_name
+                ",
+                array('v_project_name' => $projectName)
+            );
+    
+            if (empty($projectData)) {
+                throw new \LorisException("There is no project called: $projectName");
+            }
+    
+            $project = new Project();
+    
+            $project->projectId         = $projectData['projectId'];
+            $project->projectName       = $projectData['name'];
+            $project->recruitmentTarget = (integer) $projectData['recruitmentTarget'];
+    
+            self::$instances[$projectName] = $project;
+        }
+        
+        return self::$instances[$projectName];
+    }
+
+
+    /**
+     * This does the work of creating a new Project
+     *
+     * @param string  $projectName       The project name
+     * @param integer $recruitmentTarget The number of expected participants
+     *
+     * @note If the recruitmentTarget is missing or null, this project
+     * recruitmentTarget will be the sum of all its subprojects recruitmentTarget
+     * see: getRecruitmentTarget
+     *
+     * @return Project the Project object
+     * @throws DatabaseException | LorisException
+     */
+    static public function createNew($projectName, $recruitmentTarget = null)
+    {
+
+        // TODO :: Add not null constraint to the iname column of the Project table
+        if (empty($projectName)) {
+            throw new \LorisException('ProjectName can\'t be null');
+        }
+
+        $factory = \NDB_Factory::singleton();
+        // TODO :: Add unique constraint to the name column of the Project table
+        $projectId = $factory->database()->pselectOne(
+            "
+            SELECT 
+              projectId
+            FROM
+              Project
+            WHERE
+              name = :v_project_name
+            ",
+            array('v_project_name' => $projectName)
+        );
+        if (!empty($projectId)) {
+            throw new \LorisException('The project $projectName already exists');
```

Comment:
The variable $projectName won't get expanded here, since it's inside of single quotes. This one should be doubled quotes so that variables get interpolated.